spec/api/comment_thread_spec.rb

```diff
@@ -594,6 +594,24 @@ def thread_result(id, params)
         comment.endorsement.should == nil
         check_thread_result_json(nil, changed_thread, parse(last_response.body))
       end
+      it "update information of comment thread and mark thread as read" do
+        thread = CommentThread.first
+        comment = thread.comments.first
+        comment.endorsed = true
+        comment.endorsement = {:user_id => "42", :time => DateTime.now}
+        comment.save
+        put "/api/v1/threads/#{thread.id}", body: "new body", title: "new title", commentable_id: "new_commentable_id", thread_type: "question", read: true
+        last_response.should be_ok
+        changed_thread = CommentThread.find(thread.id)
+        changed_thread.body.should == "new body"
+        changed_thread.title.should == "new title"
+        changed_thread.commentable_id.should == "new_commentable_id"
+        changed_thread.thread_type.should == "question"
+        comment.reload
+        comment.endorsed.should == false
```

Comment:
Why is endorsed now set to false?

And where in this test does it check that the thread is now marked to "read"?